### ISSUE TITLE
[java] UnusedPrivateMethod false positive when using @MethodSource on a @Nested test

### ISSUE THREAD

**Issue by mdagcilar** [pmd/pmd#4975]
**Affects PMD Version:** 
7.0.0

**Rule:** 
UnusedPrivateMethod

Please provide the rule name and a link to the rule documentation:
https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod

**Description:**
I'm using java 21, with maven-pmd-plugin 3.22.0 (should be using the default PMD 7.0.0)

i am geting a false positive result for UnusedPrivateMethod when a private method is being used as part of @ParameterizedTest @MethodSource argument. 

The method is only used as part of the @MethodSource annotation.

I've removed all business sensitive context out of the test, but otherwise this is the exact structure of our test. 

**Code Sample demonstrating the issue:**

```java
class BaseTest {
    @Nested
    class ExampleTest {

        @ParameterizedTest
        @MethodSource("getStrings")
        void exampleTestUsingPrivateMethod(
                List<String> strings) {
            // insert code
        }

        private static Stream<Arguments> getStrings() {
            Stream<Arguments> tests = Stream.of(
                    Arguments.of(List.of("TEST", "TEST_1")),
                    Arguments.of(List.of("TEST_2", "TEST_3"))
            );

            return Stream.of(tests)
                    .reduce(Stream::concat)
                    .orElseGet(Stream::empty);
        }
    }
}
```

**Expected outcome:**

I have been using this rule against this block of code in prior version to PMD 7.0.0 and it has been working (i.e. not complaining)

PMD reports a violation at line 

```
Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'getStrings()'..

```

However, the method is used 

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Maven

**Comment by punkratz312** [comment]
We are still encountering false positives with plugin version 3.25.0. We prefer not to use the ignore file workaround and are hoping for a permanent fix.

<img width="724" alt="image" src="https://github.com/user-attachments/assets/bd86a348-c773-4929-8976-62162ae5c9fb">

**Comment by jsotuyod** [comment]
@punkratz312 please, feel free to open a new issue providing a failing sample.

The cases on this original issue have been covered

**Comment by punkratz312** [comment]
Hi Juan,

I noticed some strange behavior and hope you can address it. We identified 80 issues thanks to the great tool, which is fantastic. However, there are some downsides that need attention.

One key issue is that a specific rule is not functioning properly. This is a major reason why we wanted to use PMD and move away from relying on the `final` keyword. Unfortunately, the tool is not detecting actual bugs, even though the example provided was taken directly from the Java documentation.

The rule in question is: **[AvoidReassigningParametersRule](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#avoidreassigningparameters)**

Additionally, it's not detecting unused parameters in one particular method, which could be a clue. The rule related to this is: **[UnusedFormalParameter](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedformalparameter)**

[WARNING] 550 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'get6(UUID)'.
[WARNING] 555 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'greet(String)'.
[WARNING] 567 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'feet(String)'.
[WARNING] 567 Rule:UnusedFormalParameter Priority:3 Avoid unused method parameters such as 'name'.


I hope you can reproduce this behavior locally by adding these code snippets:

```java
public void get2() {
}

void get3() {
}

void get4(String s) {
}

String get5(UUID s) {
    return s.toString();
}

private String get6(UUID s) {
    s = null; // avoidreassigningparameters
    return s.toString();
}

private void greet(String name) {
    name = "";  // avoidreassigningparameters
    name = null;   // avoidreassigningparameters
    name = name.trim(); // NPE
    System.out.println("Hello " + name);

    // Preferred
    String trimmedName = name.trim();
    System.out.println("Hello " + trimmedName);
}

private void feet(String name) {
}

**Comment by punkratz312** [comment]
<img width="686" alt="image" src="https://github.com/user-attachments/assets/5c9c5496-b4da-44be-a890-dae433475c5e">
<img width="899" alt="image" src="https://github.com/user-attachments/assets/214aa97e-9aa0-4704-8290-b02fa837ab12">
